File: src/main/java/me/tajam/jext/disc/DiscPersistentDataContainer.java
package me.tajam.jext.disc;

import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.persistence.PersistentDataContainer;
import org.bukkit.persistence.PersistentDataType;

import me.tajam.jext.JextNamespace;

public class DiscPersistentDataContainer {

  private final String IDENTIFIER = "JEXT";

  private PersistentDataContainer container;

  public DiscPersistentDataContainer(ItemMeta meta) {
    this.container = meta.getPersistentDataContainer();
  }

  public void setTitle(String value) {
    container.set(
      JextNamespace.TITLE.get(),
      PersistentDataType.STRING,
      value
    );
  }

  public String getTitle() {
    return container.get(
      JextNamespace.TITLE.get(),
      PersistentDataType.STRING
    );
  }

  public void setAuthor(String value) {
    container.set(
      JextNamespace.AUTHOR.get(),
      PersistentDataType.STRING,
      value
    );
  }

  public String getAuthor() {
    return container.get(
      JextNamespace.AUTHOR.get(),
      PersistentDataType.STRING
    );
  }

  public void setNamespaceID(String value) {
    container.set(
      JextNamespace.NAMESPACE_ID.get(),
      PersistentDataType.STRING,
      value
    );
  }

  public String getNamespaceID() {
    return container.get(
      JextNamespace.NAMESPACE_ID.get(),
      PersistentDataType.STRING
    );
  }

  public void setIdentifier() {
    container.set(
      JextNamespace.IDENTIFIER.get(),
      PersistentDataType.STRING,
      IDENTIFIER
    );
  }

  public boolean checkIdentifier() {
    final String value = container.get(
      JextNamespace.IDENTIFIER.get(),
      PersistentDataType.STRING
    );
    return value.equals(IDENTIFIER);
  }

}

File: src/main/java/me/spartacus04/jext/disc/DiscPersistentDataContainer.kt
package me.spartacus04.jext.disc

import me.spartacus04.jext.JextNamespace
import org.bukkit.inventory.meta.ItemMeta
import org.bukkit.persistence.PersistentDataContainer
import org.bukkit.persistence.PersistentDataType

/**
 * This class is used to store the disc data in the item meta.
 *
 * @constructor Creates a new DiscPersistentDataContainer object
 *
 * @param meta The item meta
 */
class DiscPersistentDataContainer(meta: ItemMeta?) {
    private val id = "JEXT"
    private val container: PersistentDataContainer

    init {
        container = meta!!.persistentDataContainer
    }

    /**
     * Sets and gets the disc title from the item meta.
     */
    var title: String?
        get() = container.get(
            JextNamespace.TITLE(),
            PersistentDataType.STRING
        )
        set(value) {
            container.set(
                JextNamespace.TITLE(),
                PersistentDataType.STRING,
                value!!
            )
        }

    /**
     * Sets and gets the disc author from the item meta.
     */
    var author: String?
        get() = container.get(
            JextNamespace.AUTHOR(),
            PersistentDataType.STRING
        )
        set(value) {
            container.set(
                JextNamespace.AUTHOR(),
                PersistentDataType.STRING,
                value!!
            )
        }

    /**
     * Sets and gets the disc namespace from the item meta.
     */
    var namespaceID: String?
        get() = container.get(
            JextNamespace.NAMESPACE_ID(),
            PersistentDataType.STRING
        )
        set(value) {
            container.set(
                JextNamespace.NAMESPACE_ID(),
                PersistentDataType.STRING,
                value!!
            )
        }

    /**
     * Sets a JEXT identifier in the item meta.
     */
    fun setIdentifier() {
        container.set(
            JextNamespace.IDENTIFIER(),
            PersistentDataType.STRING,
            id
        )
    }

    /**
     * Checks if the item meta contains a JEXT identifier.
     *
     * @return A boolean value that represents whether the item meta contains a JEXT identifier or not
     */
    fun checkIdentifier(): Boolean = container.get(
        JextNamespace.IDENTIFIER(),
        PersistentDataType.STRING
    ) == id
}